evidence/pool.go

```diff
@@ -128,19 +129,27 @@ func (evpool *EvidencePool) MarkEvidenceAsCommitted(height int64, evidence []typ
 
 	// remove committed evidence from the clist
 	maxAge := evpool.State().ConsensusParams.EvidenceParams.MaxAge
-	evpool.removeEvidence(height, maxAge, blockEvidenceMap)
-
+	evpool.removeEvidence(
+		height,
```

Comment:
I think if we pass in the Header to MarkEvidenceAsCommitted instead of just the height (or maybe the timestamp, along with the height, without the full header), we can pass it in here too and avoid the need to get the LastBlockTime from the state ... then we don't need the Time field in those consensus msgs.

Reply:
True, but what about `checkSendEvidenceMessage` in evidenceReactor? where we decide whenever we should send an evidence to a peer based on its state.

Reply:
another alternative is to pass blockstore and load block header?

```go
maxAge := evR.evpool.State().ConsensusParams.EvidenceParams.MaxAge
	// peerLastBlockTime := blockStore.LoadBlockMeta(peerHeight).Header.Time
	if peerLastBlockTime.Sub(ev.Time()) > maxAge {
		// evidence is too old, skip
		// NOTE: if evidence is too old for an honest peer,
		// then we're behind and either it already got committed or it never will!
		evR.Logger.Info("Not sending peer old evidence", "peerHeight", peerHeight, "evHeight", evHeight, "maxAge", maxAge, "peer", peer)
		return nil, false
	}
```